src-self-hosted/libc_installation.zig

```diff
@@ -378,11 +374,13 @@ pub const LibCInstallation = struct {
         };
 
         for (searches) |search| {
-            result_buf.shrink(0);
+            var result_buf = std.ArrayList(u8).init(allocator);
+            defer result_buf.deinit();
```

Comment:
This is less efficient use of memory here. Before it only made one byte buffer and resized it in the loop; now it allocates and frees a fresh one every loop iteration. Why not port the code directly using ArrayList?

Reply:
Depends on your allocator
  - For smart allocators that use a freelist, they will essentially do this automatically
  - For dumber allocator, each time through the loop may have a different sized allocation and they probably don't do `realloc` efficiently, so they end up doing multiple allocations anyway!

In general I have a preference to reduce the scope of variables as much as possible.